lib/client-options.js

```diff
@@ -220,15 +258,14 @@ function ifUndefined3(v1, v2, v3) {
 }
 
 function getTimestamp(client, userOptions, defaultValue) {
-  var value = defaultValue;
   if (typeof userOptions.timestamp !== 'undefined') {
-    value = userOptions.timestamp;
+    return userOptions.timestamp;
   }
-  else if (client.controlConnection.protocolVersion > 2 && client.options.policies.timestampGeneration) {
-    // Use the timestamp generator
-    value = client.options.policies.timestampGeneration.next(client);
+  var timestampGenerator = client.options.policies.timestampGeneration;
+  if (types.protocolVersion.supportsTimestamp(client.controlConnection.protocolVersion) && timestampGenerator) {
```

Comment:
Big fan of having functions for checking support instead of basing off the version number 👍